server/utils/personUtils.test.ts

```diff
@@ -0,0 +1,70 @@
+import personUtils from './personUtils'
+import { prisonerFactory } from '../testutils/factories'
+
+describe('personUtils', () => {
+  describe('personFromPrisoner', () => {
+    describe('when all fields are present on a Prison Offender Search "Prisoner"', () => {
+      it('returns a "Person" with those fields', () => {
+        const prisoner = prisonerFactory.build({
+          firstName: 'Del',
+          lastName: 'Hatton',
+          prisonerNumber: 'ABC1234',
+          dateOfBirth: '1971-07-05',
+          gender: 'Male',
+          religion: 'Christian',
+          ethnicity: 'White',
+          imprisonmentStatusDescription: 'Concurrent determinate',
+          prisonName: 'HMP Hewell',
+        })
+
+        expect(personUtils.personFromPrisoner(prisoner)).toEqual({
+          name: 'Del Hatton',
+          prisonNumber: 'ABC1234',
+          dateOfBirth: '5 July 1971',
+          ethnicity: 'White',
+          gender: 'Male',
+          religionOrBelief: 'Christian',
+          setting: 'Custody',
+          currentPrison: 'HMP Hewell',
+          sentenceType: 'Concurrent determinate',
+        })
+      })
+    })
+
+    describe('name', () => {
+      it("formats the person's name in title case", () => {
+        const prisoner = prisonerFactory.build({ firstName: 'DEL', lastName: 'HATTON' })
+
+        expect(personUtils.personFromPrisoner(prisoner).name).toEqual('Del Hatton')
+      })
+    })
+
+    describe('when fields are missing on a Prison Offender Search "Prisoner"', () => {
+      it('returns a "Person" with those "Not entered" for those fields', () => {
+        const prisoner = prisonerFactory.build({
+          firstName: 'Del',
+          lastName: 'Hatton',
+          prisonerNumber: 'ABC1234',
+          dateOfBirth: '1971-07-05',
+          gender: 'Male',
+          religion: undefined,
+          ethnicity: undefined,
```

Comment:
Is this what a missing field will look like? The model is written as though the keys might not be there: if they would always be `undefined` I think the `?` is redundant?

Reply:
I've been using `undefined` here to represent the keys being missing as I wanted to use the factory here. In the sense that under the hood, it's the same thing. Do you think it'd be better to not use the factory here and create a new object myself, or create a factory function e.g. `withMissingFields` that explicitly creates an object without those keys?

Reply:
I think this is okay. I was more wondering whether the model needed the `?`. Sounds like the key might not be there, but yeah, essentially that means it's undefined